### PR TITLE
Add BackgroundRunner to HelloOboe

### LINKED ISSUE #960: Add example of calling AAudio from a background thread
If multiple threads are opening and closing stream then there can be race conditions that result in a crash. #953

The use of a single threaded audio "manager" can be used to avoid this.
The Java code can send start/stop messages to the thread when the app goes through its lifecycle, or in response to user input.
Error callbacks can occur when a device is disconnected. The callback can send a command to this thread.
The thread can have delays so that the app has time to respond to device changes from both Java and the error callback and then decide what to do.

Maybe we could put this in LiveEffect or Hello Oboe.

### LINKED ISSUE #1538: Use background thread when calling Oboe from examples
Calls to AudioFlinger can hang for 5-10 seconds if opening, starting, stopping, closing or doing other things that hit the audio service.

That can result in an ANR if called from the nUI thread.
We should move those calls to an AsyncTask or command thread.

We also need to document this.

### PR DESCRIPTION
**Pull request by philburk**
Call audio in background to prevent ANRs.
If AudioServer has crashed it can cause the UI thread to time out.

Fixes #960
Fixes #1538

### CODE REVIEW

File: samples/hello-oboe/src/main/java/com/google/oboe/samples/hellooboe/MainActivity.java [hunk 1]
```diff
@@ -65,6 +67,64 @@ public class MainActivity extends Activity {
...
 
+    /** Commands for background thread. */
+    private static final int WHAT_START = 100;
+    private static final int WHAT_STOP = 101;
+    private static final int WHAT_SET_DEVICE_ID = 102;
+    private static final int WHAT_SET_AUDIO_API = 103;
+    private static final int WHAT_SET_CHANNEL_COUNT = 104;
+
+    private class MyBackgroundRunner extends BackgroundRunner {
+        int audioApi;
```
Comment (robertwu1) [inline samples/hello-oboe/src/main/java/com/google/oboe/samples/hellooboe/MainActivity.java:78]: Should we explicitly define default values for these?

**Review by robertwu1 — APPROVED** [review]
Looks good to me. I left one minor nit about defining some variables.

File: samples/hello-oboe/src/main/java/com/google/oboe/samples/hellooboe/MainActivity.java (hunk 1 above)
Comment (philburk) [reply samples/hello-oboe/src/main/java/com/google/oboe/samples/hellooboe/MainActivity.java:78]: Java always initializes variables to zero so we typically do not init them. But I will document that the default zero values correspond to the UNSPECIFIED values for the three parameters.

**Comment by philburk** [comment]
Regarding whether to quit() the Looper thread, Christopher Tate says  not to:
https://groups.google.com/g/android-platform/c/_jsUqXi6DKM/m/9I_8U_0wtk0J

> Speaking very generally, never quit() your looper threads.  That method exists mostly for historical and testing
> reasons.  In Real Life™, I recommend that you continue to reuse the same looper thread(s) for the life of the
> process rather than creating/quitting them.

I will make this a Singleton so we only create one.

**Pull request merged**